Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/TanHelper.kt

```diff
@@ -0,0 +1,33 @@
+package de.rki.coronawarnapp.util
+
+import de.rki.coronawarnapp.ui.submission.TanConstants.MAX_LENGTH
+import java.security.MessageDigest
+import java.util.Locale
+
+object TanHelper {
+    private const val VALID_CHARACTERS = "23456789ABCDEFGHJKMNPQRSTUVWXYZ"
+
+    fun isChecksumValid(tan: String): Boolean {
+        if (tan.trim().length != MAX_LENGTH)
+            return false
+        val subTan = tan.substring(0, MAX_LENGTH - 1).toUpperCase(Locale.getDefault())
```

Comment:
`Locale.getDefault()` is risky, as lowercasing may be largely locale dependent. Famous example is Turkish locale (quite often used in Germany). When you uppercase a lowercase "i" there you get "İ" (dotted I), which is definitely not what you expect. Gladly, you can be happy as "I" is not an allowed character for Teletans.
To be language neutral `Locale.ROOT` should be used.

Reply:
Good to know. Thank you!